hudi-utilities/src/test/java/org/apache/hudi/utilities/testutils/sources/AbstractDFSSourceTestBase.java

```diff
@@ -189,4 +205,63 @@ public void testReadingFromSource() throws IOException {
         Option.empty(), Long.MAX_VALUE);
     assertEquals(10101, fetch6.getBatch().get().count());
   }
+
+  @Test
+  public void testCleanUpSourceAfterCommit() throws IOException {
+    dfs.mkdirs(new Path(dfsRoot));
+    TypedProperties props = new TypedProperties();
+    props.setProperty("hoodie.deltastreamer.source.dfs.clean.mode", "delete");
+    // use synchronous clean to be able to verify immediately
+    props.setProperty("hoodie.deltastreamer.source.dfs.clean.numThreads", "0");
+    SourceFormatAdapter sourceFormatAdapter = new SourceFormatAdapter(prepareDFSSource(props));
+    assertEquals(Option.empty(),
+        sourceFormatAdapter.fetchNewDataInAvroFormat(Option.empty(), Long.MAX_VALUE).getBatch());
```

Comment:
you can move the lines from 211 to 218 to a separate private method and reuse in both tests. So that main test looks lean and easy to read. 
Also, add no-op tests as well.